sqlmesh/core/config/connection.py

```diff
@@ -222,6 +274,38 @@ def init(cursor: duckdb.DuckDBPyConnection) -> None:
 
         return init
 
+    def create_engine_adapter(self, register_comments_override: bool = False) -> EngineAdapter:
+        """Checks if another engine adapter has already been created that shares a catalog that points to the same data
+        file. If so, it uses that same adapter instead of creating a new one. As a result, any additional configuration
+        associated with the new adapter will be ignored."""
+        data_files = set((self.catalogs or {}).values())
+        if self.database:
+            data_files.add(self.database)
```

Comment:
Hm, does this work with Motherduck? Database is not a file in case of Motherduck. is it?

Reply:
It would be a path formatted as `md:{db_name}?motherduck_token={token}` (both the db name and token being optional} and automatically handled by the duckdb client as a MotherDuck URL due to the prefix. Do we want to apply the "either `database` or `catalog` but not both" validation from the existing `DuckDBConnectionConfig` implementation?

Reply:
Sorry, I'm a bit confused. The `md:` prefix is added as part of the `_static_connection_kwargs` method. Here the `database` value doesn't contain it. Wouldn't it be treated as a file then?

Reply:
oh that's true, I guess we need to add `md:` when database is supplied as part of `catalogs`.